java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java

```diff
@@ -433,6 +483,52 @@ private static VectorSchemaRoot getRoot(final ResultSet data, final BufferAlloca
     return new VectorSchemaRoot(singletonList(dataVector));
   }
 
+  private static VectorSchemaRoot getTypeInfoRoot(CommandGetXdbcTypeInfo request, ResultSet typeInfo,
+                                                  final BufferAllocator allocator)
+      throws SQLException {
+    Objects.requireNonNull(allocator, "BufferAllocator cannot be null.");
+
+    VectorSchemaRoot root = VectorSchemaRoot.create(Schemas.GET_TYPE_INFO_SCHEMA, allocator);
+
+    Map<FieldVector, String> mapper = new HashMap<>();
```

Comment:
should this be a static variable someplace?

Reply:
No, because it is using the FieldVector that are not static. They are inputs.

Unfortunately, this is being used throughout the code. Because of this we've created a task to refactor it and make it more elegant.

Is it ok for us to do this refactor in a different PR? Since this will involve changes in other parts of the code.